horaedb/metric_engine/src/compaction/picker.rs

```diff
@@ -28,7 +37,105 @@ impl TimeWindowCompactionStrategy {
         Self { segment_duration }
     }
 
-    pub fn pick_candidate(&self, _ssts: Vec<SstFile>) -> Option<Task> {
-        todo!()
+    pub fn pick_candidate(
+        &self,
+        ssts: Vec<SstFile>,
+        expire_time: Option<Timestamp>,
+    ) -> Option<Task> {
+        let (uncompacted_files, expired_files) =
+            Self::find_uncompacted_and_expired_files(ssts, expire_time);
+        debug!(
+            "uncompacted_files: {:?}, expired_files: {:?}",
+            uncompacted_files, expired_files
+        );
+
+        let buckets = self.get_buckets(&uncompacted_files);
+        let compact_files = self.get_compact_files(buckets);
+
+        let expireds = expired_files
+            .into_iter()
+            .map(|f| f.id())
+            .collect::<Vec<_>>();
+
+        if compact_files.is_empty() && expireds.is_empty() {
+            return None;
+        }
+
+        let task = Task {
+            inputs: compact_files,
+            expireds,
+        };
+        debug!("Pick compaction task: {:?}", task);
+
+        Some(task)
+    }
+
+    ///  Group files of similar timestamp into buckets.
+    fn get_buckets(&self, files: &[SstFile]) -> HashMap<i64, Vec<FileId>> {
+        let mut buckets: HashMap<i64, Vec<FileId>> = HashMap::new();
+        for f in files {
+            let (left, _) = self.get_window_bounds(f.meta().time_range.end);
+
+            let bucket_files = buckets.entry(left).or_default();
+
+            bucket_files.push(f.id());
```

Comment:
FileId is not suitable here, we need to mutate sst inner state, so a Task should hold `SstFile`